PhysicsTools/PatAlgos/plugins/PATIsolatedTrackProducer.cc

```diff
@@ -777,6 +869,44 @@ void pat::PATIsolatedTrackProducer::getCaloJetEnergy(const PolarLorentzVector& p
   }
 }
 
+void pat::PATIsolatedTrackProducer::getAssociatedCaloEnergy(const PolarLorentzVector& p4,
+                                                            const EBRecHitCollection& EBRecHits,
+                                                            const EERecHitCollection& EERecHits,
+                                                            const HBHERecHitCollection& HBHERecHits,
+                                                            float& eEM,
+                                                            float& eHad) const {
+  eEM = 0.;
+  for (const auto& hit : EBRecHits) {
+    if (insideCone(p4, hit.detid(), associatedCaloEnergy_DR_))
+      eEM += hit.energy();
+  }
+  for (const auto& hit : EERecHits) {
+    if (insideCone(p4, hit.detid(), associatedCaloEnergy_DR_))
+      eEM += hit.energy();
+  }
+
+  eHad = 0.;
+  for (const auto& hit : HBHERecHits) {
+    if (insideCone(p4, hit.detid(), associatedCaloEnergy_DR_))
+      eHad += hit.energy();
+  }
+}
+
+bool pat::PATIsolatedTrackProducer::insideCone(const PolarLorentzVector& p4, const DetId& id, const double dR) const {
+  if (!caloGeometry_.isValid() || !caloGeometry_->getSubdetectorGeometry(id) ||
+      !caloGeometry_->getSubdetectorGeometry(id)->getGeometry(id)) {
+    throw cms::Exception("FatalError") << "Failed to access geometry for DetId: " << id.rawId();
+    return false;
+  }
+
+  const GlobalPoint& idPosition = caloGeometry_->getSubdetectorGeometry(id)->getGeometry(id)->getPosition();
+  if (idPosition.mag() < 0.01)
+    return false;
+
+  math::XYZVector idPositionRoot(idPosition.x(), idPosition.y(), idPosition.z());
+  return deltaR(p4, idPositionRoot) < dR;
```

Comment:
This can also get simplified by using `deltaR2` instead: just pass as argument the square of the `dR`
The *_DR parameters could be replaced by some *_DR2 ones, and as such the same simplification could be applied also (I think) in all other cases in which deltaR is used in this producer.

Reply:
deltaR2 are used in our changes of the producer.   *_DR_ parameters are kept for convenience, explicit multiplication is made for _DR_ parameters while comparing to dR2.  Other cases from original code in which deltaR is used is left 
for now.